packages/core/src/application.ts

```diff
@@ -293,128 +49,94 @@ export class Application extends Context {
-   * app.route(route);
+   * app.servers([
+   *  RestServer,
+   *  GRPCServer,
+   * ]);
+   * // Creates a binding for "servers.RestServer" and a binding for
+   * // "servers.GRPCServer";
    * ```
    *
-   * @param route The route to add.
+   * @param {Constructor<Server>[]} ctors An array of Server constructors.
+   * @memberof Application
    */
-  route(route: RouteEntry): Binding;
-
-  route(
-    routeOrVerb: RouteEntry | string,
-    path?: string,
-    spec?: OperationObject,
-    controller?: ControllerClass,
-    methodName?: string,
-  ): Binding {
-    if (typeof routeOrVerb === 'object') {
-      const r = routeOrVerb;
-      return this.bind(`routes.${r.verb} ${r.path}`).to(r);
-    }
-
-    if (!path) {
-     throw new AssertionError({
-       message: 'path is required for a controller-based route',
-     });
-    }
-
-    if (!spec) {
-     throw new AssertionError({
-       message: 'spec is required for a controller-based route',
-     });
-    }
-
-    if (!controller) {
-     throw new AssertionError({
-       message: 'controller is required for a controller-based route',
-      });
-    }
-
-    if (!methodName) {
-      throw new AssertionError({
-        message: 'methodName is required for a controller-based route',
-      });
-    }
-
-    return this.route(new ControllerRoute(
-      routeOrVerb,
-      path,
-      spec,
-      controller,
-      methodName));
+  public servers<T extends Server>(ctors: Constructor<T>[]) {
+    ctors.map(ctor => this.server(ctor));
   }
 
-  api(spec: OpenApiSpec): Binding {
-    return this.bind(CoreBindings.API_SPEC).to(spec);
+  /**
+   * Retrieve the singleton instance for a bound constructor.
+   *
+   * @template T
+   * @param {Constructor<T>} ctor The constructor that was used to make the
+   * binding.
+   * @returns {Promise<T>}
+   * @memberof Application
+   */
+  public async getServer<T extends Server>(ctor: Constructor<T>): Promise<T> {
+    return await this.get(`servers.${ctor.name}`) as T;
   }
 
   /**
-   * Get the OpenAPI specification describing the REST API provided by
-   * this application.
+   * Retrieve the singleton instance for a bound constructor synchronously.
+   * Whenever possible, you should make use of the async "getServer" instead.
    *
-   * This method merges operations (HTTP endpoints) from the following sources:
-   *  - `app.api(spec)`
-   *  - `app.controller(MyController)`
-   *  - `app.route(route)`
-   *  - `app.route('get', '/greet', operationSpec, MyController, 'greet')`
+   * @template T
+   * @param {Constructor<T>} ctor
+   * @returns {T}
+   * @memberof Application
    */
-  getApiSpec(): OpenApiSpec {
-    const spec = this.getSync(CoreBindings.API_SPEC);
-
-    // Apply deep clone to prevent getApiSpec() callers from
-    // accidentally modifying our internal routing data
-    spec.paths = cloneDeep(this.httpHandler.describeApiPaths());
-
-    return spec;
+  public getServerSync<T extends Server>(ctor: Constructor<T>): T {
```

Comment:
Considering that this is used primarily in tests, and that async functions are cheap to write, is this sync helper necessary at all?

Reply:
It was mostly to assist with the current tests; I didn't want to risk breaking them by rewriting them as async in addition to the existing changes I'm making.

Once we've got a solid foundation for the changes, I can strip this function out and re-write those tests to be async.